clients/src/main/java/org/apache/kafka/common/Uuid.java

```diff
@@ -51,15 +49,22 @@ public Uuid(long mostSigBits, long leastSigBits) {
         this.leastSignificantBits = leastSigBits;
     }
 
+    private static Uuid unsafeRandomUuid() {
+        java.util.UUID jUuid = java.util.UUID.randomUUID();
+        return new Uuid(jUuid.getMostSignificantBits(), jUuid.getLeastSignificantBits());
+    }
+
     /**
      * Static factory to retrieve a type 4 (pseudo randomly generated) UUID.
+     *
+     * This will not generate a UUID equal to 0, 1, or one whose string representation starts with a dash ("-")
      */
     public static Uuid randomUuid() {
-        java.util.UUID uuid = java.util.UUID.randomUUID();
-        while (uuid.equals(METADATA_TOPIC_ID_INTERNAL) || uuid.equals(ZERO_ID_INTERNAL)) {
-            uuid = java.util.UUID.randomUUID();
+        Uuid uuid = unsafeRandomUuid();
+        while (uuid.equals(METADATA_TOPIC_ID) || uuid.equals(ZERO_UUID) || uuid.toString().startsWith("-")) {
+            uuid = unsafeRandomUuid();
```

Comment:
You're right we will need to support leading "-" indefinitely since ZK clusters or existing KRaft clusters may have this type of clusterId. I left `Uuid#fromString` unchanged so we don't break compatibility with any extant cluster IDs.

Agree that we should change our examples in the docs and perhaps add a note that older cluster IDs may have this problem and require the `=`